test/ibmq/test_basic_server_paths.py

```diff
@@ -0,0 +1,129 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests that hit all the basic server endpoints using both a public and premium provider."""
+
+from datetime import datetime, timedelta
+
+from qiskit.compiler import assemble, transpile
+from qiskit.providers.jobstatus import JobStatus
+from qiskit.test.reference_circuits import ReferenceCircuits
+
+from qiskit.providers.ibmq import least_busy
+from ..decorators import requires_providers
+from ..ibmqtestcase import IBMQTestCase
+from ..utils import most_busy_backend, cancel_job
+
+
+class TestBasicServerPaths(IBMQTestCase):
+    """Test the basic server endpoints using both a public and premium provider."""
+
+    @classmethod
+    @requires_providers
+    def setUpClass(cls, providers):
+        # pylint: disable=arguments-differ
+        super().setUpClass()
+        cls.providers = providers  # Dict[str, AccountProvider]
+        cls._qc = ReferenceCircuits.bell()
+        cls.seed = 73846087
+
+    def test_job_submission(self):
+        """Test submitting a job."""
+        for _desc, provider in self.providers.items():
+            with self.subTest(provider=provider):
+                backend = least_busy(provider.backends())
+                circuit = transpile(self._qc, backend, seed_transpiler=self.seed)
+                qobj = assemble(circuit, backend, shots=1)
+                job = backend.run(qobj, validate_qobj=True)
+
+                # Fetch the results.
+                result = job.result()
+                self.assertTrue(result.success)
+
+                # Fetch the qobj.
+                qobj_downloaded = job.qobj()
+                self.assertEqual(qobj_downloaded, qobj.to_dict())
+
+    def test_job_status_and_properties(self):
+        """Test the status and properties of a job."""
+        for _desc, provider in self.providers.items():
+            with self.subTest(provider=provider):
+                backend = most_busy_backend(provider)
+                circuit = transpile(self._qc, backend, seed_transpiler=self.seed)
+                qobj = assemble(circuit, backend, shots=1)
+                job = backend.run(qobj, validate_qobj=True)
+
+                self.assertTrue(job.status())
+                self.assertIsNotNone(job.properties())
+                cancel_job(job, verify=True)
+
+    def test_retrieving_jobs(self):
+        """Test retrieving jobs with filters."""
+        backend_name = 'ibmq_qasm_simulator'
+        past_month = datetime.now() - timedelta(days=30)
+        past_month_str = past_month.strftime('%Y-%m-%dT%H:%M:%S.%fZ')
+        db_filter = {
+            'backend.name': backend_name,
+            'status': {'regexp': '^ERROR'}
+        }
```

Comment:
You now have a dependency that the test is only useful if there were failed jobs in the past month. Since this is only to test basic endpoints, it doesn't seem that this dependency is necessary.